### PR TITLE
fix: User Mentioning when prepending the username with >

### LINKED ISSUE #28846: Unable to highlight a user when prepending the username with a >
### Description:

When I prepend a username with `>`, the user will not be highlighted.

### Steps to reproduce:

Suppose you have a user with the username `test1`.

1. Write the message `test > @test1`.

### Expected behavior:

The user will be highlighted.

### Actual behavior:

The user will not be highlighted:

image

### Server Setup Information:

- Version of Rocket.Chat Server: 6.1.1
- Deployment Method: Docker
- Number of Running Instances: 1
- MongoDB Version: 6.0.5

### Client Setup Information

The issue seems to be client-agnostic. I just verified it with:

- Browser Version:
  - Chromium Version 112.0.5615.49 (Offizieller Build) Arch Linux (64-Bit)
  - Firefox 111.0.1 (64-bit)
- Operating System: Arch Linux

### Additional context

The issue only occurs when the `>` is immediately followed by the username.

What does not work:
- `test > @test1`
- `> @test1`

What works:
 - `> test @test1`
 - `test > test @test1`
 
image

### PR DESCRIPTION
**Pull request by ayush3160**
## Proposed changes (including videos or screenshots)

This pull request fixes the irregular behavior of mentioning the user when prepending the username with ">" symbol.

https://user-images.githubusercontent.com/89914602/231296116-c057e093-7f3d-4e49-bb29-05a54d7baef4.mp4

## Issue(s)

fix #28846 
## Steps to test or reproduce

1. Type test > @any-username.
2. Notice it is getting mentioned now.

### CODE REVIEW

**Comment by ayush3160** [comment]
@hugocostadev Can you please review this PR.

**Comment by hugocostadev** [comment]
Some tests are failing due to the changes, please, take a look

**Comment by ayush3160** [comment]
@hugocostadev , Please have a look.

**Review by hugocostadev — CHANGES_REQUESTED** [review]
I don't think that it's good to check the specific char `>`
Let's think, why `>` ?  Is there a reason for that?  

I was checking here and the `>` char it's often used for quotes, maybe the purpose of that Regex it's to accept mentions inside quotes, the same way that it's done with room mentions... That said, check how the room regex is done and let's compare it with the user's mention, there is a difference between those. 

If we change the user mention regex to match the room regex and test it we will see that this fixes the issue without creating a conditional operation in the `getUserMentions` method

What do you think?  
If you agree and test it locally you can accept my suggestions and change `userMentionRegex` function like this:

```js
	get userMentionRegex() {
		return new RegExp(`(^|\\s|>)@(${this.pattern}(@(${this.pattern}))?(:([0-9a-zA-Z-_.]+))?)`, 'gm');
	}
```

Output: 
image

**Comment by ayush3160** [comment]
@hugocostadev , Thanks for the detailed explanation and help. I applied your suggestion and also tested , it is working fine with '>' char also, Please have a look.

Also let me know if we should add a unit test case for this particular case.

Thanks.

File: apps/meteor/app/mentions/lib/MentionsParser.js [hunk 1]
```diff
@@ -115,7 +115,12 @@ export class MentionsParser {
 		});
 
 	getUserMentions(str) {
-		return (str.match(this.userMentionRegex) || []).map((match) => match.trim());
+		return (str.match(this.userMentionRegex) || []).map((match) => {
+			if (match[0] === '>') {
+				return match.slice(1).trim();
+			}
+			return match.trim();
+		});
```
[the remark this answers is not in the document]
Comment (ayush3160) [reply apps/meteor/app/mentions/lib/MentionsParser.js]: Done

**Review by hugocostadev — APPROVED** [review]

**Review by hugocostadev — APPROVED** [review]

**Pull request merged**